pkg/registry/apis/query/query.go

```diff
@@ -400,3 +423,13 @@ func (r responderWrapper) Error(err error) {
 
 	r.wrapped.Error(err)
 }
+
+// Checks if the request only contains a single query and not expression.
+func alertQueryWithoutExpression(req parsedRequestInfo) bool {
+	headers := req.Requests[0].Headers
+	_, exist := headers[models.FromAlertHeaderName]
+	if exist && len(req.Requests) == 1 && len(req.Requests[0].Request.Queries) == 1 && len(req.Expressions) == 0 {
```

Comment:
We already do part of this check in the switch case. I do it here again in case we move the code around in the future.

Reply:
should this be done a couple lines up on 429 for headers?

Reply:
Yes, moved things around to adhere to it.